src/plugins/registration.go

```diff
@@ -153,23 +156,43 @@ func (r *OneTimeRegistration) areDataplaneSoftwareDetailsReady() error {
 func (r *OneTimeRegistration) registerAgent() {
 	var details []*proto.NginxDetails
 
-	for _, proc := range r.processes {
-		// only need master process for registration
-		if proc.IsMaster {
-			nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
-			details = append(details, nginxDetails)
-			// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
-			_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
-			if err != nil {
-				log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
+	backoffSetting := backoff.BackoffSettings{
+		InitialInterval: softwareDetailsOperationInterval,
+		MaxInterval:     softwareDetailsOperationInterval,
+		MaxElapsedTime:  nginxStartMaxWaitTime,
+		Jitter:          backoff.BACKOFF_JITTER,
+		Multiplier:      backoff.BACKOFF_MULTIPLIER,
+	}
+
+	findMaster := func() error {
+		for _, proc := range r.processes {
+			// only need master process for registration
+			if proc.IsMaster {
+				nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
+				details = append(details, nginxDetails)
+				if len(details) > 0 {
+					// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
+					_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
+					if err != nil {
+						log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
```

Comment:
I know that this line isn't originally your code, but I'm wondering if we should actually be returning the error here instead of merely logging it 🤔 

@oliveromahony, @dhurley, @Dean-Coakley, @aphralG thoughts?

Reply:
I believe the reason we dont return an error here is because we dont want to stop registration just because we cant read the config. If the config is invalid it should not stop the agent from performing other tasks like reporting the status of NGINX is that makes sense.